src/alerts/time_series.py

```diff
@@ -8,24 +8,150 @@
 import matplotlib.pyplot as plt
 import pandas as pd
 from alerts.anomaly import AnomalyDetector
+from alerts.dataclasses import TimeSeries, PredictionInterval, Event, LimitInterval
 
+from abc import abstractmethod
+from abc import ABC
 
-class AnomalyTS():
 
-    QUANTILE = 1.625
+def identity(x: Any) -> Any:
+    return x
+
+
+class AnomalyTS(ABC):
+
+    @abstractmethod
+    def prediction_interval() -> List[PredictionInterval]:
+        pass
+
+    @abstractmethod
+    def plot(self) -> None:
+        pass
+
+
+class AnomalyMeanTS(AnomalyTS):
+
+    Z_SCORE = 1.625
     WINDOW = 30
 
     def __init__(self,
-                 T: List[Tuple[Any, int]],
-                 parameters: Dict[str, Any]):
-        """Time series anomaly
-
-        Args:
-            T (List[Tuple[int, int]]): represents data point containing a
-        timestamp and an associating value.
-            parameters (Dict[str, Any]): set of hyper-parameters that help to
-                compute the prediction intervals.
+                 ts: TimeSeries,
+                 z_score: float = Z_SCORE,
+                 window: int = WINDOW,
+                 flat_ts: bool = False):
+        self.z_score = z_score
+        self.window = window
+        self.flat_ts = flat_ts
+        self.ts = ts
+
+    def _init_prediction_interval(self) -> List[PredictionInterval]:
+        """Init the prediction intervals
+
+        Returns:
+            List[PredictionInterval]: The list contains
+                the events pre window frame
         """
+        prediction_interval = list()
+        for event in self.ts.events[:self.window]:
+            prediction_interval.append(
+                PredictionInterval(
+                    t=event.t,
+                    v=np.sqrt(event.v) if self.flat_ts else event.v,
+                    lb=None,
+                    ub=None
+                )
+            )
+        return prediction_interval
+
+    def _compute_limits(
+            self,
+            current_window_frame: List[Event]) -> LimitInterval:
+
+        values = [e.v for e in current_window_frame]
+        if self.flat_ts:
+            values = np.sqrt(self.ts.values)
+
+        mean_value = np.mean(values)
+        std_value = np.std(values)
+        lower_bound = mean_value - self.z_score * std_value
+        upper_bound = mean_value + self.z_score * std_value
+
+        limits = LimitInterval(
+            lb=lower_bound,
+            ub=upper_bound
+        )
+
+        return limits
+
+    def prediction_interval(self) -> List[PredictionInterval]:
+
+        prediction_interval = self._init_prediction_interval
+        future_events = self.ts.events[self.window:]
+
+        for i, (date, value) in enumerate(future_events):
+
+            current_window_frame = self.ts.events[i: self.window + i]
+
+            limits = self._compute_limits(current_window_frame)
+            lower_bound = 0 if limits.lower_bound < 0 else limits.upper_bound
```

Comment:
If the under limit of lower bound is 0 must be validated in dataclass. 
And not will be `else lower_bound`?
```suggestion
            lower_bound = 0 if limits.lower_bound < 0 else limits.lower_bound
```

Reply:
@quimdt  Could be yes